plugins/Disintegrator/Disintegrator.cpp

```diff
@@ -0,0 +1,279 @@
+/*
+ * Disintegrator.cpp
+ *
+ * Copyright (c) 2019 Lost Robot <r94231@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "Disintegrator.h"
+
+#include "embed.h"
+#include "lmms_math.h"
+#include "plugin_export.h"
+
+extern "C"
+{
+
+Plugin::Descriptor PLUGIN_EXPORT disintegrator_plugin_descriptor =
+{
+	STRINGIFY(PLUGIN_NAME),
+	"Disintegrator",
+	QT_TRANSLATE_NOOP("pluginBrowser", "A delay modulation effect for creating very aggressive digital distortion."),
+	"Lost Robot <r94231@gmail.com>",
+	0x0100,
+	Plugin::Effect,
+	new PluginPixmapLoader("logo"),
+	NULL,
+	NULL
+} ;
+
+}
+
+
+
+DisintegratorEffect::DisintegratorEffect(Model* parent, const Descriptor::SubPluginFeatures::Key* key) :
+	Effect(&disintegrator_plugin_descriptor, parent, key),
+	m_disintegratorControls(this)
+{
+	for (int a = 0; a < 200; ++a)
+	{
+		for (int b = 0; b < 2; ++b)
+		{
+			m_inBuf[b].push_back(0);
+		}
+	}
+}
+
+
+
+
+DisintegratorEffect::~DisintegratorEffect()
+{
+}
+
+
+
+
+bool DisintegratorEffect::processAudioBuffer(sampleFrame* buf, const fpp_t frames)
+{
+	if (!isEnabled() || !isRunning ())
+	{
+		return false;
+	}
+
+	double outSum = 0.0;
+	const float d = dryLevel();
+	const float w = wetLevel();
+	
+	const ValueBuffer * lowCutBuf = m_disintegratorControls.m_lowCutModel.valueBuffer();
+	const ValueBuffer * highCutBuf = m_disintegratorControls.m_highCutModel.valueBuffer();
+	const ValueBuffer * amountBuf = m_disintegratorControls.m_amountModel.valueBuffer();
+	const ValueBuffer * typeBuf = m_disintegratorControls.m_typeModel.valueBuffer();
+	const ValueBuffer * freqBuf = m_disintegratorControls.m_lowCutModel.valueBuffer();
+
+	sample_rate_t sample_rate = Engine::mixer()->processingSampleRate();
```

Comment:
Names with camel cases are preferred because it's consistent with other parts of the code.
```suggestion
	sample_rate_t sampleRate = Engine::mixer()->processingSampleRate();
```